ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java

```diff
@@ -148,6 +150,70 @@ public static SourceBuilderV1 instance() {
     );
   }
 
+  KTableHolder<Windowed<GenericKey>> buildWindowedTable(
+      final RuntimeBuildContext buildContext,
+      final SourceStep<KTableHolder<Windowed<GenericKey>>> source,
+      final ConsumedFactory consumedFactory,
+      final MaterializedFactory materializedFactory,
+      final PlanInfo planInfo
+  ) {
+    final PhysicalSchema physicalSchema = getPhysicalSchema(source);
+
+    final Serde<GenericRow> valueSerde = getValueSerde(buildContext, source, physicalSchema);
+
+    final WindowInfo windowInfo;
+
+    if (source instanceof WindowedTableSource) {
+      windowInfo = ((WindowedTableSource) source).getWindowInfo();
+    } else {
+      throw new IllegalArgumentException("Expected a version of WindowedTableSource");
+    }
+
+    final Serde<Windowed<GenericKey>> keySerde = SourceBuilderUtils.getWindowedKeySerde(
+        source,
+        physicalSchema,
+        buildContext,
+        windowInfo
+    );
+
+    final Consumed<Windowed<GenericKey>, GenericRow> consumed = buildSourceConsumed(
+        source,
+        keySerde,
+        valueSerde,
+        AutoOffsetReset.EARLIEST,
+        buildContext,
+        consumedFactory
+    );
+
+    final String stateStoreName = tableChangeLogOpName(source.getProperties());
+
+    final Materialized<Windowed<GenericKey>, GenericRow, KeyValueStore<Bytes, byte[]>> materialized
+        = buildWindowedTableMaterialized(
```

Comment:
Let's inline this method -- the method definition is longer than the implementation.